packages/hyperion-autologging/src/ALIReactFlowlet.ts

```diff
@@ -0,0 +1,142 @@
+ * the body of the method has access to flowlet.
+ * For class components, we store the flowlet in the `this` object.
+ * For function components, we have to keep that value and close on it (we could use useRef)
+ */
+
+  const IS_FLOWLET_SETUP_PROP = 'isFlowletSetup';
+
+  type ComponentWithFlowlet = React.Component<any> & {
+    _flowlet?: ALFlowlet | null | undefined
+  };
+
+  IReactComponent.onReactClassComponentIntercept.add(shadowComponent => {
+    const methods = [
+      shadowComponent.render,
+      shadowComponent.componentWillMount,
+      shadowComponent.componentDidMount,
+      shadowComponent.componentWillReceiveProps,
+      shadowComponent.shouldComponentUpdate,
+      shadowComponent.componentWillUpdate,
+      shadowComponent.componentDidUpdate,
+      shadowComponent.componentWillUnmount,
+      shadowComponent.componentDidCatch,
+    ];
+
+
+    methods.forEach(method => {
+      if (method.getData<boolean>(IS_FLOWLET_SETUP_PROP)) {
+        return;
+      }
+      method.setData(IS_FLOWLET_SETUP_PROP, true);
+
+      if (method === shadowComponent.render) {
+        method.onArgsObserverAdd(function (this: ComponentWithFlowlet) {
+          /**
+              * We cannot call hooks inside of the class components.
+              * Also, a class component can only have a single context assigned to it.
+              * So, we use this internal api/hack to get the value.
+              * This is ok, because this is a best effort and anyways most people are
+              * migrating away from class components.
+              */
+          const ctx = (ALSurfaceContext as any)._currentValue; // Unofficial internal value
+          const activeFlowlet = ctx?.flowlet;
+          if (activeFlowlet) {
+            flowletManager.push(activeFlowlet);
+            this._flowlet = activeFlowlet;
+          }
+        });
+      } else {
+        method.onArgsObserverAdd(function (this: ComponentWithFlowlet) {
+          const activeFlowlet = this._flowlet;
+          if (activeFlowlet) {
+            flowletManager.push(activeFlowlet);
+          }
+        });
+      }
+
+      method.onValueObserverAdd(function (this: ComponentWithFlowlet) {
+        const activeFlowlet = this._flowlet;
+        if (activeFlowlet) {
+          flowletManager.pop(activeFlowlet);
+        }
+      });
+    });
+  });
+
+  IReactComponent.onReactFunctionComponentIntercept.add(
+    fi => {
+      if (!fi.getData<boolean>(IS_FLOWLET_SETUP_PROP)) {
+        fi.setData(IS_FLOWLET_SETUP_PROP, true);
+
+        let activeFlowlet: ALFlowlet | undefined | null;
+        fi.onArgsObserverAdd(_props => {
+          const ctx = useALSurfaceContext();
+          activeFlowlet = ctx.flowlet;
+          if (activeFlowlet) {
+            flowletManager.push(activeFlowlet)
+          }
+        });
+        fi.onValueObserverAdd(() => {
+          if (activeFlowlet) {
+            if (__DEV__) {
+              assert(
```

Comment:
Is there a way to get these to log to `FBLogger` in WWW? It would be good to track when these happen and see if we can plug the gaps

Reply:
Yes, I had devised the plan for doing this later, but didn't get a chance to expose it. 
Basically, we should be able to replace the default logger here: 

https://github.com/facebook/hyperion/blob/main/packages/global/src/assert.ts#L14-L17